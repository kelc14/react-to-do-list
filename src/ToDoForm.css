@import url("https://fonts.googleapis.com/css2?family=Roboto+Slab:wght@300;500&display=swap");

.ToDoForm-btn {
  background-color: rgb(152, 238, 152);
  border-color: rgb(152, 238, 152);
  border-radius: 4px;
  font-family: "Roboto Slab", serif;
  font-weight: 300;
}
.ToDoForm-input {
  font-family: "Roboto Slab", serif;
  font-weight: 300;
}
