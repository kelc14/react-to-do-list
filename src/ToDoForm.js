import React, { useState } from "react";
import "./ToDoForm.css";

const ToDoForm = ({ addItem }) => {
  const INITIAL_STATE = { name: "" };
  const [formData, setFormData] = useState(INITIAL_STATE);

  /** Send {name} to parent
   *    & clear form. */

  const handleSubmit = (evt) => {
    evt.preventDefault();
    addItem(formData);
    setFormData(INITIAL_STATE);
  };

  /** Update local state w/curr state of input elem */

  const handleChange = (evt) => {
    const { name, value } = evt.target;
    setFormData((fData) => ({
      ...fData,
      [name]: value,
    }));
  };

  /** Render:
   *
   */
  return (
    <div className="ToDoForm" onSubmit={handleSubmit}>
      <form className="ToDoForm-form">
        <label htmlFor="name">Task:</label>
        <input
          id="name"
          type="text"
          name="name"
          onChange={handleChange}
          value={formData.name}
          className="ToDoForm-input"
        />

        <button className="ToDoForm-btn">Add to List</button>
      </form>
    </div>
  );
};

export default ToDoForm;
