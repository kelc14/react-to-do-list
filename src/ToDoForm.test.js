import { render, fireEvent } from "@testing-library/react";
import ToDoForm from "./ToDoForm";

// smoke test
it("renders without crashing", function () {
  render(<ToDoForm />);
});

// snapshot test
it("matches snapshot", function () {
  const { asFragment } = render(<ToDoForm />);
  expect(asFragment()).toMatchSnapshot();
});
