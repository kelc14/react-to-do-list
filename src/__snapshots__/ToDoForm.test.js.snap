// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ToDoForm"
  >
    <form
      class="ToDoForm-form"
    >
      <label
        for="name"
      >
        Task:
      </label>
      <input
        class="ToDoForm-input"
        id="name"
        name="name"
        type="text"
        value=""
      />
      <button
        class="ToDoForm-btn"
      >
        Add to List
      </button>
    </form>
  </div>
</DocumentFragment>
`;
